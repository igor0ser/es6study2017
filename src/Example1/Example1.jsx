import React, { Component } from 'react';

class Exapmle1 extends Component {
  constructor(props) {
    super(props);
    const { value1, value2, value3, value4 } = props;

    this.state = {
      counter: 0,
      valuesSumm: value1 + value2 + value3 + value4
    };

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    console.log('counter before - ', this.state.counter);
    // functional setState
    /*
    this.setState((state) => ({
      counter: state.counter + 1
    }));
    */

    // non-functional
    this.setState({
      counter: this.state.counter + 1
    });

    console.log('counter after - ', this.state.counter);
  }

  render() {
    const { value1, value2, value3, value4 } = this.props;
    const { counter, valuesSumm } = this.state;

    return (
      <section onClick={this.handleClick}>
        <h3>Example 1</h3>
        <ul>
          <li>Counter: {counter}</li>
          <li>Values summ: {valuesSumm}</li>
          <li>Value 1: {value1}</li>
          <li>Value 2: {value2}</li>
          <li>Value 3: {value3}</li>
          <li>Value 4: {value4}</li>
        </ul>
      </section>
    );
  }
}

Exapmle1.defaultProps = {
  value1: 1,
  value2: 2,
  value3: 3,
  value4: 4
};

export default Exapmle1;
