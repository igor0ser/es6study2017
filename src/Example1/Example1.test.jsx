import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Example1 from './Example1';

describe('Example1', () => {
    let comp;

    beforeEach(() => {
        comp = shallow(
            <Example1
                value1={10}
                value2={20}
            >
                dummy
            </Example1>
        );
    });

    it('should render proper props and children is not passed', () => {
        expect(comp.find('li')).toHaveLength(7);
        expect(comp.find('li').at(6).text()).toEqual('dummy');
    });

    it('should match snapshot', () => {
        expect(toJson(comp)).toMatchSnapshot();
    });

    it('should increase counter on click', () => {
        comp.find('section').simulate('click');
        comp.find('section').simulate('click');
        comp.find('section').simulate('click');

        expect(comp.find('li').at(0).text()).toEqual('Counter: 3');
    });
});
