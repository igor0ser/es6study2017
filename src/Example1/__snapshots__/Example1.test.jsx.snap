// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example1 should match snapshot 1`] = `
<section
  onClick={[Function]}
>
  <h3>
    Example 1
  </h3>
  <ul>
    <li>
      Counter: 
      0
    </li>
    <li>
      Values summ: 
      37
    </li>
    <li>
      Value 1: 
      10
    </li>
    <li>
      Value 2: 
      20
    </li>
    <li>
      Value 3: 
      3
    </li>
    <li>
      Value 4: 
      4
    </li>
    <li>
      dummy
    </li>
  </ul>
</section>
`;
